code/go/0chain.net/chaincore/smartcontractinterface/auth.go

```diff
@@ -0,0 +1,13 @@
+package smartcontractinterface
+
+import (
+	"0chain.net/core/common"
+)
+
+func AuthorizeWithOwner(funcName string, hasAccess func() bool) error {
```

Comment:
This is common logic that we use in different places, we might want something more complex and hasAccess can differ from time to time (we can use several fields, some check to authorized maps and so on) it's like a pointcut in aspect programming, which is a common way to implement authorization logic. 

In your example, you can't  have different equality logic for ownerId and ClientId (imagine you want to call isEqual() instead of '==' ) or some other logic like less than (stake < 1000) or contains in the map. It should give us  agility in the future and creates near zero boilerplate. Maybe it'll be better to rename `AuthorizeWithOwner ` to `Authorize`